src/core/electrostatics_magnetostatics/p3m_gpu_cuda.cu

```diff
@@ -744,6 +744,17 @@ void p3m_gpu_init(int cao, const int mesh[3], double alpha) {
     }
 
     if ((p3m_gpu_data_initialized == 0) && (p3m_gpu_data.mesh_size > 0)) {
+#if defined(__HIPCC__) and not defined(__CUDACC__)
+      // rocFFT gives wrong P3M results for mesh sizes whose prime factors are
+      // not 2, 3 or 5. So we check for the other supported prime factors and
+      // throw an error if they are used with HIP.
+      if (mesh[0] % 7 == 0 || mesh[0] % 11 == 0 || mesh[0] % 13 == 0 ||
+          mesh[1] % 7 == 0 || mesh[1] % 11 == 0 || mesh[1] % 13 == 0 ||
+          mesh[2] % 7 == 0 || mesh[2] % 11 == 0 || mesh[2] % 13 == 0) {
+        throw std::string("Mesh size not supported");
```

Comment:
```throw std::runtime_error("Mesh size not supported");```

Reply:
Why do you throw ```std::string```?

Reply:
copy-and-paste error from 23 lines further down... I just noticed that there are a few more places in Espresso that make the same mistake and should probably be changed to a `runtime_error`.